tests/virt_autotest/esxi_open_vm_tools.pm

```diff
@@ -0,0 +1,214 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Run open-vm-tools testing against VMware ESXi
+# Maintainer: Nan Zhang <nan.zhang@suse.com>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use virt_autotest::esxi_utils;
+use Time::Local;
+
+my $VM_POWERED_ON  = 'Powered on';
+my $VM_POWERED_OFF = 'Powered off';
+
+sub run {
+    assert_script_run('rpmquery open-vm-tools');
+    systemctl('status vmtoolsd');
+    systemctl('status vgauthd');
+
+    my $vm_name = console('svirt')->name;
+    my $vm_id   = esxi_vm_get_vmid($vm_name);
+    my $vm_ip   = esxi_vm_public_ip($vm_id);
+
+    record_info('Guest Name', $vm_name);
+    record_info('Guest ID',   $vm_id);
+    record_info('Guest IP',   $vm_ip);
+
+    # Power Manangement Tests
+    do_power_management_tests($vm_id, $vm_ip);
+    # Networking Tests
+    do_networking_tests($vm_id, $vm_ip);
+    # Clock Sync Tests
+    do_clock_sync_tests($vm_name, $vm_id, $vm_ip);
+}
+
+sub check_power_state {
+    my ($vm_id, $vm_ip, $power_action, $if_pingable, $if_vm_state, $expect_vm_state) = @_;
+
+    if (esxi_vm_power_getstate($vm_id) =~ m/$if_vm_state/) {
+        esxi_vm_power_cycle($vm_id, $power_action);
+        wait_for_vm_bootup($vm_ip, $if_pingable);
+    } else {
+        die "Not the correct VM state for guest $power_action.";
+    }
+    die "Failed to $power_action the guest." if (!esxi_vm_power_getstate($vm_id) =~ m/$expect_vm_state/);
+}
+
+sub wait_for_vm_bootup {
+    my $vm_ip       = shift;         # vm ip address
+    my $if_pingable = shift;         # 1 or 0
+    my $times       = shift // 6;    # the number of ping execution
+    my $count       = shift // 3;    # ping waits for the count
+    my $interval    = shift // 5;    # wait secs between each ping
+    my $cmd;
+
+    if ($if_pingable) {
+        $cmd = "ping -c$count -i$interval $vm_ip";
+    } else {
+        $cmd = "! ping -c$count -i$interval $vm_ip";
+    }
+
+    while ($times && console('svirt')->run_cmd($cmd)) {
+        $times--;
+    }
+}
+
+sub login_vm_console {
+    reset_consoles;
+    select_console('sut');
+    assert_screen('grub2', 90);
+    send_key 'ret';
+    select_console('root-console', timeout => 180);
+}
+
+sub set_guest_time {
```

Comment:
Although you only use this function after login_vm_console, this functions seems can be used everywhere. To be honest, the function name does not match with what it is supposed to do.

Reply:
How about the function name 'set_random_time'?

Reply:
It is more like get_random_time ?

Reply:
This function invokes 'date' command with '-s' option, it's actually for setting a time. So I gave the prefix 'set_' for the name.
-s, --set=STRING           set time described by STRING

Reply:
That's OK.